Source/WebCore/accessibility/AXObjectRareData.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "AXCoreObject.h"
+#include "AccessibilityObject.h"
+#include <wtf/TZoneMalloc.h>
+
+namespace WebCore {
+
+DECLARE_COMPACT_ALLOCATOR_WITH_HEAP_IDENTIFIER(AXObjectRareData);
+class AXObjectRareData {
+    WTF_MAKE_NONCOPYABLE(AXObjectRareData);
+    WTF_MAKE_COMPACT_TZONE_ALLOCATED(AXObjectRareData);
+public:
+    AXObjectRareData() = default;
+
+    unsigned rowCount() { return m_rows.size(); }
```

Comment:
Should we have a flag for "is a table" or "has table attributes" and assert it here?

Or alternatively maybe we just need to assert that in AccessibilityObject when we fetch this from rare data.

Reply:
Hmm, yeah it feels like it's the responsibility of the rare-data owner to know and properly use it. I'll add the isTable ASSERT on the AccessibilityRenderObject / AccessibilityNodeObject / AccessibilityObject side where the rare data is accessed.